packages/remix-react/components.tsx

```diff
@@ -961,9 +961,14 @@ export function useFormAction(
 ): string {
   let { id } = useRemixRouteContext();
   let path = useResolvedPath(action);
+  let location = useLocation();
   let search = path.search;
   let isIndexRoute = id.endsWith("/index");
 
+  if (action === ".") {
```

Comment:
This should check if `action === undefined` instead. If the action is explicitly set to `.` it will be resolved to strip the search, which is the expected behavior.

We only want to use the location's search if the action is left undefined in the Form. This means we need to remove the default value of the action in `FormImpl` as well.

See https://github.com/remix-run/remix/issues/3133#issuecomment-1128776268

Reply:
Got it!